### ISSUE TITLE
feat: Add predict_proba() for RandomForestClassifier

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#72]
## Summary

Add probability prediction support to `RandomForestClassifier` to enable confidence scores in downstream applications.

## Motivation

The Organizational Intelligence Plugin (OIP) uses aprender's `RandomForestClassifier` for defect categorization. Currently, the classifier only returns class labels, not probabilities:

```rust
// Current: returns label index only
let prediction = classifier.predict(&features)?;

// Needed: return probability distribution over classes
let probabilities = classifier.predict_proba(&features)?;
// probabilities: Vec<f32> where probabilities[i] = P(class_i | features)
```

**Impact**: OIP currently uses a placeholder confidence of 0.75 for all predictions, reducing the utility of the classifier for:
- Ranking predictions by confidence
- Multi-label prediction (top-N categories)
- Filtering low-confidence classifications

## Proposed API

```rust
impl RandomForestClassifier {
    /// Predict class probabilities for input features
    ///
    /// Returns probability distribution over classes based on
    /// vote proportions across trees in the forest.
    pub fn predict_proba(&self, features: &Matrix<f32>) -> Result<Matrix<f32>> {
        // For each sample, count votes per class across all trees
        // Return normalized vote counts as probabilities
    }
}
```

## Implementation Notes

RandomForest probability estimation:
1. Each tree votes for a class
2. `P(class_i) = votes_for_class_i / n_trees`
3. This is the standard sklearn approach

## References

- scikit-learn RandomForestClassifier.predict_proba: https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html#sklearn.ensemble.RandomForestClassifier.predict_proba
- OIP ml_trainer.rs:449 - placeholder confidence workaround

## Acceptance Criteria

- [ ] `predict_proba()` method added to `RandomForestClassifier`
- [ ] Returns `Matrix<f32>` with shape `(n_samples, n_classes)`
- [ ] Probabilities sum to 1.0 for each sample
- [ ] Unit tests with known probability distributions
- [ ] Documentation with examples

**Comment by noahgift** [comment]
Implemented `predict_proba()` for RandomForestClassifier.

Returns a Matrix<f32> with shape (n_samples, n_classes) where each row sums to 1.0.

Example:
```rust
let mut rf = RandomForestClassifier::new(100);
rf.fit(&x_train, &y_train)?;

let proba = rf.predict_proba(&x_test);
// proba.get(0, 1) = probability of class 1 for sample 0
```
